hardware/VirtualThermostat.cpp

```diff
@@ -0,0 +1,682 @@
+#include "stdafx.h"
+#include "../main/Helper.h"
+#include "../main/Logger.h"
+#include "../main/SQLHelper.h"
+#include "../main/RFXtrx.h"
+
+#include "VirtualThermostat.h"
+
+#include "../main/localtime_r.h"
+#include "../main/mainworker.h"
+#include "../notifications/NotificationHelper.h"
+#include <stdarg.h>     
+
+std::string AVALAIBLE_MODE = "Eco,Conf,Frost,Off";
+
+std::string GetDeviceValue(const char* FieldName, const char* Idx)
+{
+	auto result = m_sql.safe_query("SELECT %s from DeviceStatus WHERE (ID == %s )", FieldName, Idx);
+	if (!result.empty())
+		return  result[0][0];
+	else
+		return  "";
+}
+
+void VirtualThermostat::CircularBuffer::Clear()
+{
+	for (int i = 0; i < m_Size; i++)m_Value[i] = 0;
+	m_index = 0;
+}
+VirtualThermostat::CircularBuffer::CircularBuffer(int pSize)
```

Comment:
`unsigned int` would be better here and also in member variables and for cycles.

Reply:
ok.done !